trestle/core/ssp_io.py

```diff
@@ -0,0 +1,330 @@
+
+        Args:
+            control_id: The control_id to use.
+
+        Returns:
+            A markdown blob as a string.
+        """
+        if not self._resolved_catalog:
+            raise TrestleError('Cannot get control statement, set resolved catalog first.')
+
+        writer = ControlIOWriter()
+        control = self._catalog_interface.get_control(control_id)
+        group_id, group_title, _ = self._catalog_interface.get_group_info_by_control(control.id)
+
+        control_lines = writer.get_control_statement(control, group_title)
+
+        return self._build_tree_and_adjust(control_lines, level)
+
+    def get_control_part(self, control_id: str, part_name: str, level: int):
+        """Get control part with given name."""
+        control_part = self._catalog_interface.get_control_part_prose(control_id, part_name)
+
+        md_list = self._write_str_with_header(
+            f'Control Part: {part_name} for control: {control_id}', control_part, level
+        )
+        return self._build_tree_and_adjust(md_list.split('\n'), level)
+
+    def get_fedramp_control_tables(self, control_id: str, level: int) -> str:
+        """Get the fedramp metadata as markdown tables.
+
+        The fedramp metadata has the following elements:
+        - Responsible roles field
+        - Parameter values table
+        - Implementation status field
+        - Control origination field
+
+        Returns:
+            tables as one coherent markdown blob.
+        """
+        resp_roles_table = self.get_responsible_roles_table(control_id, level)
+        params_values = self._parameter_table(control_id, level)
+        impl_status = self.get_fedramp_implementation_status(control_id, level)
+        control_orig = self.get_fedramp_control_origination(control_id, level)
+
+        final_output = ''
+        if resp_roles_table:
+            final_output += resp_roles_table
+        if params_values:
+            final_output += '\n' + params_values
+        if impl_status:
+            final_output += '\n' + impl_status
+        if control_orig:
+            final_output += '\n' + control_orig
+        return final_output
+
+    def get_responsible_roles_table(self, control_id: str, level: int) -> str:
+        """
+        For each role id - if the role exists in metadata use the title as what gets printed in the roles table.
+
+        If not (for now) warn and use the role-id as the printed text.
+        """
+        if self._ssp is None:
+            raise TrestleError('Cannot get responsible roles, SSP is not set.')
+
+        for impl_requirement in self._ssp.control_implementation.implemented_requirements:
+            if impl_requirement.control_id == control_id:
+                if impl_requirement.responsible_roles:
+                    role_ids = []
+                    for resp_role in impl_requirement.responsible_roles:
+                        role_ids.append(resp_role.role_id)
```

Comment:
When I create roles from the markdown header and put them in the json ssp, I need to make them ncname compatible, which means spaces get converted to _.  There is no great way to do this, but for the final markdown it might be good just to convert all _ to spaces at this point so they look good.